_posts/2023-02-15-my-first-patch-set.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: post
+title:  "My First Patch Set"
+date:   2023-02-15 11:57:26 -0300
+categories: jekyll update
+---
+
+Hello everyone! In this Post i will describe my experience writing my first patch set to the Linux
+Kernel, more specifically in the [Direct Rendering Manager (DRM)][wikipedia-drm] subsystem.
+
+## Why I'm Doing This
+
+Besides the fun of contributing to a project that millions of people use, This is part of my
+initiation on the [GSoC project Idea about Increasing Code Coverage on the DRM
+code][xorg-gsoc-project-idea], a project that I pretend to do on this year.
+
+## Patch Set Idea
+
+The idea is to decrease number of warnings generated when compiling the AMD GPU drivers with `W=1`.
+This is a pretty simple task to start working on, as is not needed an specific hardware to test it.
+
+Sometimes this work could be viewed as not useful, but it is a much needed job that sometimes is put
+aside on behalf of more urgent problems. It makes the code less error prone, more readable and
+simpler, helping others on developing.
+
+## Set Up
+
+To start I added the `drm-misc` tree as it was recommend.
+
+```bash
+git remote add drm-misc git://anongit.freedesktop.org/drm/drm-misc
+```
+
+And created a local branch based on the `drm-misc-next` branch.
+
+```bash
+git checkout drm-misc/drm-misc-next
+git checkout -b fix-warning
+```
+
+After that I used the defconfig and added the AMDGPU config as a module with the menuconfig.
+
+```bash
+make defconfig
+make menuconfig # Add AMDGPU as a module
+```
+
+## Workflow
+
+The workflow is to build and try to resolve the warnings that appear, but the kernel uses the
+`-Werror` flag so the build will stop after any warning. To speed up the process by resolving
+multiples warning in one go it is possible to add `-Wno-error` to `KCFLAGS` in the make command.
+This will compile the module and generate all the warnings presented, this can be a bit overwhelming
+as all types of warnings appear together.
```

Comment:
```suggestion
The idea is trying to resolve the warnings that appear when building the AMDGPU code. Considering that the kernel uses the `-Werror` flag, the build will stop after any warning, which can make the development process a bit tricky. To speed up the process it is possible to add `-Wno-error` to `KCFLAGS` in the `make` command. This way you will be able to resolve multiple warnings in one go.

This will compile the module and generate all the warnings, which can be a bit overwhelming as all types of warnings appear together.
```

Reply:
A suggestion for making the sentence more clear. I guess you can also mention that you are making a W=1 build.